src/services/identity/LaunchesService.ts

```diff
@@ -146,6 +165,84 @@ export class LaunchesService {
     return ok(siteId)
   }
 
+  getDNSRecords = (
+    repoName: string
+  ): ResultAsync<DnsResultsForSite, AmplifyError | SiteLaunchError> => {
+    let dnsRecords: DNSRecord[] = []
+    let siteUrl: string
+    return ResultAsync.fromPromise(this.getSiteId(repoName), () =>
+      createErrorAndLog(AmplifyError, `Failed to get siteId for ${repoName}`)
+    )
+      .andThen((siteId) => {
+        if (siteId.isOk()) {
```

Comment:
I am ok to make this change but in a separate PR as this would change the contract for other methods that is not within the scope of the PR + after we get clarification if as a team are ok with chaining constructs/if we just want to use `isOk`/ `isErr`, ie effectively returning a `Promise<Result>`

Reply:
sure - file a ticket for it! i'd hope for the ticket to be worked on soon to prevent the signature from propagating

Reply:
agreed, filed ticket [here](https://isomeropengov.atlassian.net/browse/IS-288?atlOrigin=eyJpIjoiOGJjZjgwZTkwY2QyNDc4MDgyYmU4YTQzY2JmMTY0NDAiLCJwIjoiaiJ9)